security/src/main/java/io/micronaut/security/token/TokenAuthenticationFetcher.java

```diff
@@ -37,66 +38,63 @@
 
 /**
  * Attempts to retrieve a token form the {@link HttpRequest} and if existing validated.
- * It uses the list of {@link TokenReader} and {@link TokenValidator} registered in the ApplicationContext.
+ * It uses a {@link TokenResolver} and the list of {@link TokenValidator} registered in the ApplicationContext.
  *
  * @author Sergio del Amo
  * @author Graeme Rocher
  * @since 1.0
  */
 @Singleton
 public class TokenAuthenticationFetcher implements AuthenticationFetcher {
+
     /**
      * The order of the fetcher.
      */
     public static final Integer ORDER = 0;
 
-    private static final Logger LOG = LoggerFactory.getLogger(TokenAuthenticationFetcher.class);
-
-    protected final Collection<TokenReader> tokenReaders;
     protected final Collection<TokenValidator> tokenValidators;
+    private final TokenResolver tokenResolver;
     protected final ApplicationEventPublisher eventPublisher;
 
     /**
      * @param tokenValidators The list of {@link TokenValidator} which attempt to validate the request
-     * @param tokenReaders    The list {@link TokenReader} which attempt to read the request
+     * @param tokenResolver   The {@link io.micronaut.security.token.reader.TokenResolver} which returns the first found token in the request.
      * @param eventPublisher  The Application event publiser
      */
+    @Inject
     public TokenAuthenticationFetcher(Collection<TokenValidator> tokenValidators,
-                                      Collection<TokenReader> tokenReaders,
+                                      TokenResolver tokenResolver,
                                       ApplicationEventPublisher eventPublisher) {
         this.eventPublisher = eventPublisher;
+        this.tokenResolver = tokenResolver;
         this.tokenValidators = tokenValidators;
-        this.tokenReaders = tokenReaders;
     }
 
+    /**
+     * @param tokenValidators The list of {@link TokenValidator} which attempt to validate the request
+     * @param tokenReaders    The list {@link TokenReader} which attempt to read the request
+     * @param eventPublisher  The Application event publiser
+     */
+    @Deprecated
```

Comment:
This needs documentation on how to move to the newer constructor

Reply:
I've added a java doc comment. @jameskleeh let me know if this can be merged.